distributed/locket.py

```diff
@@ -154,26 +156,24 @@ def release(self):
 class _LockFile:
     def __init__(self, path):
         self._path = path
-        self._file = None
         self._thread_lock = threading.Lock()
 
     def acquire(self, timeout=None, retry_period=None):
-        if self._file is None:
-            self._file = open(self._path, "wb")
-        if timeout is None and _lock_file_blocking_available:
-            _lock_file_blocking(self._file)
-        else:
-            _acquire_non_blocking(
-                acquire=lambda: _lock_file_non_blocking(self._file),
-                timeout=timeout,
-                retry_period=retry_period,
-                path=self._path,
-            )
+        with open(self._path, "wb") as fd:
+            if timeout is None and _lock_file_blocking_available:
+                _lock_file_blocking(fd)
+            else:
+                _acquire_non_blocking(
+                    acquire=lambda: _lock_file_non_blocking(fd),
+                    timeout=timeout,
+                    retry_period=retry_period,
+                    path=self._path,
+                )
 
     def release(self):
-        _unlock_file(self._file)
-        self._file.close()
-        self._file = None
+        if os.path.exists(self._path):
+            with open(self._path, "wb") as fd:
+                _unlock_file(fd)
```

Comment:
This seems less than ideal, but if someone has already purged the worker directory this lock file is gone and we'll throw an exception here now.

I originally included a warning but it is reraised as an error during Dask's test run and we'd have to ignore it or avoid calling `release` after the work space has been deleted.

```suggestion
        if os.path.exists(self._path):
            with open(self._path, "wb") as fd:
                _unlock_file(fd)
        else:
            warnings.warn(f"Lock at {self._path!r} no longer exists and cannot be released.", stacklevel=2)
```